external/storm-hdfs/src/main/java/org/apache/storm/hdfs/spout/HdfsSpout.java

```diff
@@ -0,0 +1,739 @@
+      LOG.info("Completed processing {}. Spout Id = {}", newFile, spoutId);
+    } catch (IOException e) {
+      LOG.error("Unable to archive completed file" + filePath + " Spout ID " + spoutId, e);
+    }
+    closeReaderAndResetTrackers();
+  }
+
+  private void markFileAsBad(Path file) {
+    String fileName = file.toString();
+    String fileNameMinusSuffix = fileName.substring(0, fileName.indexOf(inprogress_suffix));
+    String originalName = new Path(fileNameMinusSuffix).getName();
+    Path  newFile = new Path( badFilesDirPath + Path.SEPARATOR + originalName);
+
+    LOG.info("Moving bad file {} to {}. Processed it till offset {}. SpoutID= {}", originalName, newFile, tracker.getCommitPosition(), spoutId);
+    try {
+      if (!hdfs.rename(file, newFile) ) { // seems this can fail by returning false or throwing exception
+        throw new IOException("Move failed for bad file: " + file); // convert false ret value to exception
+      }
+    } catch (IOException e) {
+      LOG.warn("Error moving bad file: " + file + " to destination " + newFile + " SpoutId =" + spoutId, e);
+    }
+    closeReaderAndResetTrackers();
+  }
+
+  private void closeReaderAndResetTrackers() {
+    inflight.clear();
+    tracker.offsets.clear();
+    retryList.clear();
+
+    reader.close();
+    reader = null;
+    releaseLockAndLog(lock, spoutId);
+    lock = null;
+  }
+
+  private static void releaseLockAndLog(FileLock fLock, String spoutId) {
+    try {
+      if(fLock!=null) {
+        fLock.release();
+        LOG.debug("Spout {} released FileLock. SpoutId = {}", fLock.getLockFile(), spoutId);
+      }
+    } catch (IOException e) {
+      LOG.error("Unable to delete lock file : " +fLock.getLockFile() + " SpoutId =" + spoutId, e);
+    }
+  }
+
+  protected void emitData(List<Object> tuple, MessageId id) {
+    LOG.trace("Emitting - {}", id);
+    this.collector.emit(tuple, id);
+    inflight.put(id, tuple);
+  }
+
+  public void open(Map conf, TopologyContext context, SpoutOutputCollector collector) {
+    LOG.info("Opening HDFS Spout");
+    this.conf = conf;
+    this.commitTimer = new Timer();
+    this.tracker = new ProgressTracker();
+    this.hdfsConfig = new Configuration();
+
+    this.collector = collector;
+    this.hdfsConfig = new Configuration();
+    this.tupleCounter = 0;
+
+    // Hdfs related settings
+    if( conf.containsKey(Configs.HDFS_URI)) {
+      this.hdfsUri = conf.get(Configs.HDFS_URI).toString();
+    } else {
+      throw new RuntimeException(Configs.HDFS_URI + " setting is required");
+    }
+
+    try {
+      this.hdfs = FileSystem.get(URI.create(hdfsUri), hdfsConfig);
+    } catch (IOException e) {
+      LOG.error("Unable to instantiate file system", e);
+      throw new RuntimeException("Unable to instantiate file system", e);
+    }
+
+
+    if ( conf.containsKey(configKey) ) {
+      Map<String, Object> map = (Map<String, Object>)conf.get(configKey);
+        if(map != null) {
```

Comment:
indentation is messed up.  the if and its block should be on same level as the map variable declaration.